src/providers/seventv/SeventvEmotes.cpp

```diff
@@ -131,114 +139,44 @@ boost::optional<EmotePtr> SeventvEmotes::emote(const EmoteName &name) const
-    payload.insert("variables", variables);
+    QString url("https://api.7tv.app/v2/emotes/global");
 
-    NetworkRequest(apiUrlGQL, NetworkRequestType::Post)
+    NetworkRequest(url)
         .timeout(30000)
-        .header("Content-Type", "application/json")
-        .payload(QJsonDocument(payload).toJson(QJsonDocument::Compact))
-        .onSuccess([this](NetworkResult result) -> Outcome {
-            QJsonArray parsedEmotes = result.parseJson()
-                                          .value("data")
-                                          .toObject()
-                                          .value("search_emotes")
-                                          .toArray();
-            qCDebug(chatterinoSeventv)
-                << "7TV Global Emotes" << parsedEmotes.size();
-
-            auto pair = parseGlobalEmotes(parsedEmotes, *this->global_.get());
+        .onSuccess([this](auto result) -> Outcome {
+            auto emotes = this->emotes();
+            auto pair = parseGlobalEmotes(result.parseJsonArray(), *emotes);
             if (pair.first)
                 this->global_.set(
                     std::make_shared<EmoteMap>(std::move(pair.second)));
             return pair.first;
+            return Failure;
         })
         .execute();
 }
 
 void SeventvEmotes::loadChannel(std::weak_ptr<Channel> channel,
                                 const QString &channelId,
+                                const QString &channelDisplayName,
                                 std::function<void(EmoteMap &&)> callback,
                                 bool manualRefresh)
 {
-    qCDebug(chatterinoSeventv)
-        << "Reloading 7TV Channel Emotes" << channelId << manualRefresh;
-
-    QJsonObject payload, variables;
-
-    QString query = R"(
-        query loadUserEmotes($login: String!) {
-            user(id: $login) {
-                emotes {
-                    id
-                    name
-                    provider
-                    provider_id
-                    visibility
-                    mime
-                    owner {
-                        id
-                        display_name
-                        login
-                        twitch_id
-                    }
-                }
-            }
-        })";
+    qCDebug(chatterinoSeventv) << "[7TVEmotes] Reloading 7TV Channel Emotes"
+                               << channelId << manualRefresh;
 
-    variables.insert("login", channelId);
-
-    payload.insert("query", query.replace(whitespaceRegex, " "));
-    payload.insert("variables", variables);
-
-    qDebug() << QJsonDocument(payload).toJson(QJsonDocument::Compact);
-
-    NetworkRequest(apiUrlGQL, NetworkRequestType::Post)
-        .timeout(20000)
-        .header("Content-Type", "application/json")
-        .payload(QJsonDocument(payload).toJson(QJsonDocument::Compact))
+    NetworkRequest("https://api.7tv.app/v2/users/" + channelDisplayName +
```

Comment:
Why did you choose to use the channel's display name here? Twitch IDs are supported when querying a channel's emotes (e.g. `https://api.7tv.app/v2/users/82674227/emotes` would resolve mine)

Reply:
Didn't realize there was ID support (only looked at the docs). I will switch it to id so things are cleaner.

Reply:
Also keep in mind that [`QString::arg()`](https://doc.qt.io/qt-5/qstring.html#arg) should rather be used instead of concatenating strings together like this.
Just noticed that you've removed `apiUrlGQL` from the header file, perhaps a format string should be rather declared and used here with an arg method instead of concatenating paths together. Look at my example here: https://github.com/SevenTV/chatterino7/pull/68/files#diff-ca99a25309ab269b0aa09e72c2f777538409d73926e95d33155c20b730e6bce0L30-L32

Reply:
Sure, the arg method seems a bit more elegant. I will change to that.